src/app/ReadClient.h

```diff
@@ -305,13 +305,7 @@ class ReadClient : public Messaging::ExchangeDelegate
 
     void OnUnsolicitedMessageFromPublisher()
     {
-        // accelerate resubscription if scheduled
-        if (mIsResubscriptionScheduled)
-        {
-            ChipLogDetail(DataManagement, "%s ReadClient[%p] resubscribe on unsolicited message", __func__, this);
-            CancelResubscribeTimer();
-            OnResubscribeTimerCallback(nullptr, this);
-        }
+        TriggerResubscribeIfScheduled("unsolicited message");
```

Comment:
Could this cause a forever loop preventing a client from "no longer subscribing"? I am not sure I see all side-effects or paths that can cause a re-sub.

Reply:
Just to be clear: what this code is doing is that _if_ we decided we needed to try to resubscribe and set a timer to do it later (to avoid spamming things) then we clear the timer and try immediately, because we just got a "liveness checkin" (in the form of a data report message) from the publisher.

Once we implement actual checkin messages, those would also trigger `TriggerResubscribeIfScheduled`.

If a client no longer wishes to subscribe, it would destroy the ReadClient, or prevent a resubscribe from being scheduled using our existing APIs for this.

So in particular, if `mIsResubscriptionScheduled` is true that means the client wants the resubscribe attempt to happen.  The only thing we are changing here is _when_ it happens.